classes/Cart.php

```diff
@@ -1268,6 +1269,11 @@ public function checkAndUpdateAddresses()
             $addr = 'id_address_' . $type;
             if ($this->{$addr} != 0
                 && !Address::isValid($this->{$addr})) {
+                if ($type == 'delivery' && $this->id) {
```

Comment:
better use `Validate::isLoadedObject($this)` instead of the `$this->id` condition